purchase_order_line_payment_forecast/models/purchase_order_line.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2024 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    price_subtotal_billed = fields.Monetary(
+        compute="_compute_price_subtotal", store=True
+    )
+    price_subtotal_unbilled = fields.Monetary(
+        compute="_compute_price_subtotal", store=True
+    )
+    payment_term_id = fields.Many2one(related="order_id.payment_term_id", store=True)
+    expected_payment_date = fields.Date(
+        compute="_compute_expected_payment_date", store=True
+    )
+
+    @api.depends("price_unit", "qty_invoiced", "qty_to_invoice")
+    def _compute_price_subtotal(self):
+        for line in self:
+            if line.is_deposit:
+                sum_billed_amount_other_lines = sum(
+                    ol.price_unit * ol.qty_invoiced
+                    for ol in line.order_id.order_line
+                    if ol.id != line.id and not ol.is_deposit
+                )
+                line.price_subtotal_billed = line.price_unit
+                line.price_subtotal_unbilled = (
+                    sum_billed_amount_other_lines - line.price_unit
+                )
+
+                continue
```

Comment:
I think my initial advice wasn't valid.  This should probably do.  Please double check.

```suggestion
                line.price_subtotal_billed = line.price_unit
                line.price_subtotal_unbilled = sum(line.invoice_lines.mapped("balance")) * -1
                continue
```